numba/core/dispatcher.py

```diff
@@ -432,6 +436,10 @@ def error_rewrite(e, issue_type):
             # ignore the FULL_TRACEBACKS config, this needs reporting!
             raise e
 
+        self._types_active_call = None
```

Comment:
Shouldn't this be inside a `finally` so that the reset happens even when an error is raised?

Reply:
yes, fixed.